### PR TITLE
Z Properties

### PR DESCRIPTION
**Pull request by jnsiemer**
This PR implements useful leftovers from the `MatZq::norm` implementation. This includes
- [X] from_fmpz
- [X] distance
- [X] abs/ absolute value

including tests for `Z`.

### CODE REVIEW

File: src/integer/z/from.rs [hunk 1]
```diff
@@ -337,3 +360,21 @@ mod tests_from_modulus {
...
+
+    /// Ensure that `from_fmpz` works correctly for small and large numbers
+    #[test]
+    fn large_small_numbers() {
+        let mut mod_1 = fmpz(0);
+        unsafe { fmpz_init_set_ui(&mut mod_1, u64::MAX) };
+        let mod_2 = fmpz(0);
+
+        let _ = Z::from_fmpz(&mod_1);
+        let _ = Z::from_fmpz(&mod_2);
```
Comment (Marvin-Beckmann) [inline src/integer/z/from.rs:378]: memory leak: mod_1 has to be freed manually

File: src/integer/z/from.rs [hunk 2]
```diff
@@ -314,11 +339,9 @@ mod tests_from_modulus {
     use crate::integer_mod_q::Modulus;
     use std::str::FromStr;
 
-    /// Ensure that the `from_<type_name>` functions are available for
-    /// singed and unsigned integers of 8, 16, 32, and 64 bit length.
-    /// Tested with their maximum value.
+    /// Ensure that `from_modulus` works correctly for small and large numbers
```
Comment (Marvin-Beckmann) [inline src/integer/z/from.rs:342]: see below

File: src/integer/z/from.rs (hunk 1 above)
Comment (Marvin-Beckmann) [inline src/integer/z/from.rs:378]:
```suggestion
        let mod_1 = Z::from(u64::MAX);
        let mod_2 = Z::ZERO;

        let _ = Z::from_fmpz(&mod_1.value);
        let _ = Z::from_fmpz(&mod_2.value);
```

**Review by Marvin-Beckmann — CHANGES_REQUESTED** [review]

File: src/integer/z/from.rs (hunk 2 above)
Comment (jnsiemer) [inline src/integer/z/from.rs:342]:
```suggestion
    /// Ensure that `from_modulus` is available for small and large numbers
```

File: src/integer/z/from.rs [hunk 3]
```diff
@@ -337,3 +360,21 @@ mod tests_from_modulus {
...
     }
 }
+
+#[cfg(test)]
+mod tests_from_fmpz {
+    use super::fmpz;
+    use super::Z;
+    use flint_sys::fmpz::fmpz_init_set_ui;
+
+    /// Ensure that `from_fmpz` works correctly for small and large numbers
```
Comment (jnsiemer) [inline src/integer/z/from.rs:370]:
```suggestion
    /// Ensure that `from_fmpz` is available for small and large numbers
```

**Pull request merged**